client/state-hooks.ts

```diff
@@ -19,3 +19,29 @@ export function usePrevious<T>(value: T): T | undefined {
   })
   return ref.current
 }
+
+/**
+ * A hook to move a prop to a ref, so it can be used in `useCallback` and `useEffect` hooks, without
+ * needing to specify it in the dependency arrays of those hooks, thus reducing the amount of times
```

Comment:
This is a pretty roundabout way of saying what the purpose *actually* is, imo.

The real purpose is to make it so callbacks + effects can access the current value of the prop without needing to be re-run/re-created. The "dependencies array" part is incidental, that's just the mechanic React uses to make these decisions